draft-gruessing-moq-requirements.md

```diff
@@ -215,18 +215,35 @@ Different use cases have varying requirements with respect to the tradeoffs asso
+There should be a degree of decoupling from the underlying transport protocol and MoQ itself despite the "Q" in the name, in particular to provide future agility and prevent any potential ossification being tied to specific version(s) of dependant protocols.
+
+Many of the use cases will be deployed in contexts where web browsers are the common application runtime; thus the use of existing protocols and APIs is desireable for implementations. Support for WebTransport {{I-D.draft-ietf-webtrans-overview}} SHOULD be defined, however should not be required by all implementations or deployments.
 
 Considerations should be made clear with respect to modes where WebTransport "falls back" to using HTTP/2 or other future non-QUIC based protocol.
 
 ### Media Negotiation & Agility {#MOQ-negotiation}
 
-All entities which directly process media will have support for a variety of media codecs, consequently there SHOULD be capability in the protocol for sender and receiver to negotiate which media codecs will be used in a given session, as well as defining behaviours when one or more is not supported. Media encryption {{MOQ-media-encryption}} should also be factored in as part of the negotiation. Renegotiation SHOULD also be possible, allowing for changes in codec within an existing session. Consideration should be made if relays are also able to facilitate negotiation even if not directly processing media.
+All entities which directly process media will have support for a variety of media codecs, consequently there SHOULD be capability in the protocol for sender and receiver to negotiate which media codecs will be used in a given session, as well as defining behaviours when one or more is not supported. Media encryption {{MOQ-media-encryption}} should also be factored in as part of the negotiation. Renegotiation SHOULD also be possible, allowing for changes in codec within an existing session.
 
 The protocol SHOULD remain codec agnostic as much as possible, and should allow for new media formats and codecs to be supported without change in specification.
 
 The working group should consider if a minimum, suggestive set of codecs should be supported for the purposes of interop, however this SHOULD avoid being strict to simplify use cases and deployments that don't require certain capability e.g. telephony which may not require video codecs.
 
+## Media Data Model
+
+As the protocol will handle many different types of media, classifications, and variations when all entities describe the media a model should be defined which represents this, with a clear addressing scheme. This should factor in at least, but not limited to allow future types:
+
+Media Types
+: Video, audio, subtitles, ancillary data
+
+Classifications
+: Codec, media language, layers
+
+Variations
+: For each stream, the resolution(s), bitrate(s). Each variant should be uniquely identifiable.
```

Comment:
"Media Language" here is supposed to refer to the human language (English, Español, 日本語), which is important mostly for audio and subtitles. In reality it may need to go bigger than that for some data types. This is something that is described in a lot of existing data models to provide information for players. Maybe just "language" is enough?